x/interchainquery/keeper/keeper.go

```diff
@@ -117,32 +119,38 @@ func (k Keeper) QueryBalances(ctx sdk.Context, zone stakeibctypes.HostZone, cb C
 		bz,
 		// TODO(TEST-79) understand and use proper period
 		sdk.NewInt(25),
+		strconv.FormatInt(height, 10),
 		types.ModuleName,
 		cb,
 	)
+
+	// TODO(TEST-119) get gaia LC height, pass to height
+
 	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
 			sdk.EventTypeMessage,
 			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
 			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueQuery),
-			sdk.NewAttribute(types.AttributeKeyQueryId, GenerateQueryHash(connectionId, chainId, query_type, bz)),
+			sdk.NewAttribute(types.AttributeKeyQueryId, GenerateQueryHash(connectionId, chainId, query_type, bz, strconv.FormatInt(height, 10))),
 			sdk.NewAttribute(types.AttributeKeyChainId, chainId),
 			sdk.NewAttribute(types.AttributeKeyConnectionId, connectionId),
 			sdk.NewAttribute(types.AttributeKeyType, query_type),
-			sdk.NewAttribute(types.AttributeKeyHeight, "0"),
+			// TODO(TEST-119) set height based on gaia LC height
+			sdk.NewAttribute(types.AttributeKeyHeight, strconv.FormatInt(height, 10)),
 			sdk.NewAttribute(types.AttributeKeyRequest, hex.EncodeToString(bz)),
 		),
 	})
 	return nil
 }
 
-func (k Keeper) QueryDelegatorDelegations(ctx sdk.Context, zone stakeibctypes.HostZone, cb Callback, address string) error {
+func (k Keeper) QueryDelegatorDelegations(ctx sdk.Context, zone stakeibctypes.HostZone, cb Callback, address string, height int64) error {
 	connectionId := zone.ConnectionId
 	chainId := zone.ChainId
 
 	query_type := "cosmos.staking.v1beta1.Query/DelegatorDelegations"
 	// Get delegationAddress dynamically
 	delegationQuery := stakingtypes.QueryDelegatorDelegationsRequest{DelegatorAddr: address}
+	k.Logger(ctx).Info(fmt.Sprintf("\tabout to QueryDelegatorDelegations %s at height %d", address, height))
 	bz := k.cdc.MustMarshal(&delegationQuery)
 
```

Comment:
nit: you can probably set `heightStr := strconv.FormatInt(height, 10)` and then reference that in the two calls below